### PR TITLE
Add HCP Packer registry image metadata to builder artifacts

### PR DESCRIPTION
**Pull request by nywilken**

### CODE REVIEW

File: builder/common/artifact.go [hunk 1]
```diff
@@ -133,16 +128,32 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
...
 func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
-	metadata := make([]hcpPackerRegistryImage, 0, len(a.Amis))
 
-	for region, imageId := range a.Amis {
-		metadata = append(metadata, hcpPackerRegistryImage{
+	f := func(k, v interface{}) (*registryimage.Image, error) {
+
+		region, ok := k.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value of key in Amis map")
```
Comment (SwampDragons) [inline builder/common/artifact.go:136]:
```suggestion
			return nil, fmt.Errorf("unexpected type of key in Amis map")
```

Comment (SwampDragons) [reply builder/common/artifact.go:136]: nitpick but you're doing a type assertion here.

File: builder/common/artifact.go [hunk 2]
```diff
@@ -133,16 +128,32 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
...
+		}
+		imageId, ok := v.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value for %q in Amis map", region)
```
Comment (SwampDragons) [inline builder/common/artifact.go:140]:
```suggestion
			return nil, fmt.Errorf("unexpected type for %q in Amis map", region)
```

Comment (SwampDragons) [reply builder/common/artifact.go:140]: if it's not a string, does %q work?

**Review by SwampDragons — APPROVED** [review]
This looks great!

File: builder/common/artifact.go (hunk 2 above)
Comment (nywilken) [reply builder/common/artifact.go:140]: Oh good catch, it wont work.

File: builder/common/artifact.go [hunk 3]
```diff
@@ -133,16 +128,32 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
...
-		metadata = append(metadata, hcpPackerRegistryImage{
+	f := func(k, v interface{}) (*registryimage.Image, error) {
+
+		region, ok := k.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected type of key in Amis map")
+		}
+		imageId, ok := v.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected type for %q in Amis map", region)
```
Comment (nywilken) [inline builder/common/artifact.go:140]:
```suggestion
			return nil, fmt.Errorf("unexpected type for value in Amis map")
```

**Comment by SwampDragons** [comment]
Merge when ready 🎉

**Review by azr — APPROVED** [review]

**Pull request merged**